ablySpec/RealtimeClientChannel.swift

```diff
@@ -769,6 +876,31 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                it("with a attach callback should subscribe and call the callback when attached") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    let publishedMessage = ARTMessage(name: "foo", data: "bar")
+
+                    waitUntil(timeout: testTimeout) { done in
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Initialised))
+
+                        channel.subscribeWithAttachCallback({ errorInfo in
```

Comment:
There's already a `subscribe:cb:` method that takes a message name in its first parameter. Objective-C doesn't let you overload only by type, you need to either have different number of params, different names for them, or just take any type (`id`, the equivalent to Java's `Object`) and downcast it manually, which we don't do anywhere else and is strongly anti-idiomatic.